city_scrapers/spiders/il_sex_offender_management.py

```diff
@@ -0,0 +1,249 @@
+            )
+            meeting["status"] = self._get_status(meeting)
+            meeting["id"] = self._get_id(meeting)
+
+        yield meeting
+
+    def _cleanUpPDF(self, response):
+        """Clean up and return text string of PDF"""
+        lp = LAParams(line_margin=0.1)
+        out_str = StringIO()
+        extract_text_to_fp(BytesIO(response.body), out_str, laparams=lp)
+        pdf_text = out_str.getvalue().replace("\n", "")
+        # Remove duplicate characters not followed by lowercase (as in 5:00pm)
+        clean_text = re.sub(r"([A-Z0-9:])\1(?![a-z])", r"\1", pdf_text, flags=re.M)
+        # Remove duplicate spaces
+        clean_text = re.sub(r"\s+", " ", clean_text)
+        return clean_text
+
+    def _parse_status(self, clean_text):
+        """Check if meeting was/is cancelled."""
+        if "meeting is cancelled" in clean_text.lower():
+            return CANCELLED
+        # I went through every agenda PDF, for meetings that weren't either
+        # cancelled or moved, the agenda began with "Welcome/Roll Call"
+        if "Welcome/Roll Call" not in clean_text:
+            return CANCELLED
+        return PASSED
+
+    def _parse_start(self, clean_text):
+        """Parse start datetime as a naive datetime object."""
+        date_str = self._get_meeting_date(clean_text)
+        time_str = self._get_start_end_time(clean_text)
+
+        # if meeting is cancelled or no time is provided, default is 12:00am
+        if time_str is None:
+            return self._get_datetime_obj(date_str, "12:00am")
+        return self._get_datetime_obj(date_str, time_str[:7])
+
+    def _parse_end(self, clean_text):
+        """Parse end datetime as a naive datetime object."""
+        date_str = self._get_meeting_date(clean_text)
+        time_str = self._get_start_end_time(clean_text)
+        # if meeting is cancelled or no time is provided, default is 12:00am
+        if time_str is None:
+            return self._get_datetime_obj(date_str, "12:00am")
+        return self._get_datetime_obj(date_str, time_str[7:])
+
+    def _get_meeting_date(self, clean_text):
+        """Parse meeting agenda for 'Month Day, Year' and returns group"""
+        pattern = re.compile(
+            r"(Jan(uary)?|Feb(ruary)?|Mar(ch)?|Apr(il)?|May|Jun(e)?|"
+            r"Jul(y)?|Aug(ust)?|Sep(tember)?|Oct(ober)?|Nov(ember)?|"
+            r"Dec(ember)?)\s+\d{1,2},\s+\d{4}"
+        )
+        return pattern.search(clean_text).group()
+
+    def _get_start_end_time(self, clean_text):
+        """Parse start and end time of meeting from agenda and returns group"""
+        pattern = re.compile(
+            r"[0-2]?[0-9]:[0-9][0-9][a-zA-Z][a-zA-Z]-"
+            r"[0-2]?[0-9]:[0-9][0-9][a-zA-Z][a-zA-Z]"
+        )
+        if pattern.search(clean_text) is not None:
+            return pattern.search(clean_text).group()
+        else:
+            # in a few specific agendas there are times formatted as '9-10am'
```

Comment:
It could be quicker to try multiple datetime formats or using `dateparse` instead of doing a lot of manual cleanup here, but let me know what you think!

Reply:
I looked through all of the agendas, and for the most part (besides the video link) the PDFs have similar datetime formats! I feel comfortable with leaving it as is, but let me know otherwise.